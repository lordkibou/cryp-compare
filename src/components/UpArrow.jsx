import React, { useState, useEffect } from "react";

const ScrollButton = () => {
  const [isVisible, setIsVisible] = useState(false);

  useEffect(() => {
    const toggleVisibility = () => {
      if (window.scrollY > 300) {
        setIsVisible(true);
      } else {
        setIsVisible(false);
      }
    };

    window.addEventListener("scroll", toggleVisibility);

    return () => {
      window.removeEventListener("scroll", toggleVisibility);
    };
  }, []);

  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  return (
    <button
      className={`scroll-button ${isVisible ? "visible" : ""}`}
      onClick={scrollToTop}
    >
      <span className="arrow">&#708;</span>
    </button>
  );
};

export default ScrollButton;
