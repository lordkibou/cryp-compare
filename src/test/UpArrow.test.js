import renderer from 'react-test-renderer';
import UpArrow from '../components/UpArrow';
import { BrowserRouter } from 'react-router-dom';
//If the change is expected you can invoke Jest with jest -u to overwrite the existing snapshot.

it('Testing UpArrow Scene', () => {
  const component = renderer.create(
    <BrowserRouter>
        <UpArrow/>
    </BrowserRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});