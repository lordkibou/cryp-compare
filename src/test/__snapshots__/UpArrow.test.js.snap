// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UpArrow Scene 1`] = `
<button
  className="scroll-button "
  onClick={[Function]}
>
  <span
    className="arrow"
  >
    Ë„
  </span>
</button>
`;
